src/content/peerconnection/negotiate-timing/js/main.js

```diff
@@ -0,0 +1,232 @@
+/*
+ *  Copyright (c) 2017 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+ */
+
+'use strict';
+
+const startButton = document.getElementById('startButton');
+const callButton = document.getElementById('callButton');
+const renegotiateButton = document.getElementById('renegotiateButton');
+const hangupButton = document.getElementById('hangupButton');
+const log = document.getElementById('log');
+const videoSectionsField = document.getElementById('videoSections');
+
+callButton.disabled = true;
+hangupButton.disabled = true;
+renegotiateButton.disabled = true;
+startButton.onclick = start;
+callButton.onclick = call;
+renegotiateButton.onclick = renegotiate;
+hangupButton.onclick = hangup;
+
+let startTime;
+const localVideo = document.getElementById('localVideo');
+const remoteVideo = document.getElementById('remoteVideo');
+
+let audioTransceiver;
+let audioImpairmentAtStart = 0;
+
+localVideo.addEventListener('loadedmetadata', function() {
+  console.log(`Local video videoWidth: ${this.videoWidth}px,  videoHeight: ${this.videoHeight}px`);
+});
+
+remoteVideo.addEventListener('loadedmetadata', function() {
+  console.log(`Remote video videoWidth: ${this.videoWidth}px,  videoHeight: ${this.videoHeight}px`);
+});
+
+remoteVideo.onresize = () => {
+  console.log(`Remote video size changed to ${remoteVideo.videoWidth}x${remoteVideo.videoHeight}`);
+  console.warn('RESIZE', remoteVideo.videoWidth, remoteVideo.videoHeight);
+  // We'll use the first onsize callback as an indication that video has started
+  // playing out.
+  if (startTime) {
+    const elapsedTime = window.performance.now() - startTime;
```

Comment:
If elapsedTime is what you plan to measure, it will need to be at global scope.

Reply:
Not the only thing I want to change when doing the performance metric, I suspect (I also want per-call timings), so leaving as-is for now.